cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -723,7 +723,7 @@ predicate nodeToDefOrUse(Node node, SourceVariable sv, IRBlock bb, int i, boolea
  */
 private predicate indirectConversionFlowStep(Node nFrom, Node nTo) {
   not exists(SourceVariable sv, IRBlock bb2, int i2 |
-    nodeToDefOrUse(nTo, sv, bb2, i2, _) and
+    useToNode(bb2, i2, sv, nTo) and
```

Comment:
Based on the commit message from d020f93005b2b2bd07b1a7618d0eb02449b4ce76 I'm not sure I understand what is going on here.

Reply:
This is just a cleanup that shouldn't have any result changes. The `indirectConversionFlowStep` predicate is used as a step relation for a transitive closure in `adjustForPointerArith` where we step backwards along conversions from a pre-update node to the first node that has a def-use of use-use step (this is needed because `p + 1` isn't marked as a use of `p`, so we have to traverse backwards from `p + 1` to `p`). To check whether there is def-use of use-use information we used this conjunct on `main`:
```ql
not exists(SourceVariable sv, IRBlock bb2, int i2 |
  nodeToDefOrUse(nTo, sv, bb2, i2, _) and
  adjacentDefRead(bb2, i2, sv, _, _)
)
```
which ensures that this predicate only steps forward as long as there's no def-use or use-use information out of the node at `(bb2, i2)`.

However, since we know that we're only doing conversion steps in this predicate, we will never actually get to a node that has _def_-use information since a conversion won't step to a `StoreInstruction` (since conversions happen on register instructions, and a `StoreInstruction` produces a memory instruction).

So there's no need to handle the case where `(bb2, i2)` maps to a definition, and we can just use `useToNode` (which is one of the cases in `nodeToDefOrUse`) instead of calling `nodeToDefOrUse`.

Does that make sense?

Reply:
Makes sense.